BlueprintUICommonControls/Sources/Image.swift

```diff
@@ -137,7 +140,12 @@ extension Image {
                 } else if case .atMost(let height) = constraint.height {
                     mode = .fitHeight(height)
                 } else {
-                    mode = .useImageSize
+                    switch layoutMode {
+                    case .legacy:
+                        mode = .useImageSize
+                    case .caffeinated:
+                        mode = .infinite
```

Comment:
Could you unpack this a bit for me?

Given an unconstrained measurement space, `Image` now requests infinite layout space. Does that not change the layout behavior of images in these contexts, and therefore any container of these images?

Reply:
> Given an unconstrained measurement space, `Image` now requests infinite layout space. Does that not change the layout behavior of images in these contexts, and therefore any container of these images?

It does! Practically speaking, there's not that much impact, because the need to measure an element fully unconstrained is somewhat rare.

There's perhaps a semantic change — if you were counting on unconstrained measurement to get you something like an intrinsic size, or SwiftUI's concept of "ideal size", that's no longer possible. But it was never guaranteed to work that way, and would be inconsistent depending on what you measured.

> Or is it that `.infinite` opts them out of caffeinated layout in a way that leads to the restoration of existing behavior?

Nope, there's no way to opt out at the element level.

Reply:
Is this because of the cache hinting optimizations? I guess I would expect for the measurement modes that don't stretch the image, I'd expect to return the size of image, even in an infinite constraint.

Reply:
> Is this because of the cache hinting optimizations? 

It's required by the new layout contract, yeah.

> I guess I would expect for the measurement modes that don't stretch the image, I'd expect to return the size of image, even in an infinite constraint.

Well, the aspect fit & fill modes already fill space today, even under finite constraints. Given one finite constraint, they'll fill it, no matter how large. So the reasoning is that if the size grows continuously with the constraint, why wouldn't it grow to infinity when the constraint grows to infinity?

We could change the behavior of these modes to return the size of the image without filling space, and that would be compliant too, but it would be a much bigger breaking change to existing layouts.